### ISSUE TITLE
DLL doesn't work on Windows 10 after 1809 update

### ISSUE THREAD

**Issue by Mek7** [Portisch/USB-IR-Remote-Receiver#2]
Hello,
I was using this IR receiver for a long time in old version of Girder, with the DLL as Girder plugin. However, after latest Windows 1809 update, it stopped working. The DLL displays error "Ungultige Zeigeroperation" and IR codes are not received at all. Same behavior occurs in the demo apps (console / C#).
I dont't think you are going to fix it, just reporting. Because the DLL has no source code published, I already started making my own software that I will use instead of the old buggy girder version with the old buggy DLL. I will publish it on my website once it is done: http://mekweb.eu/?lang=en&q=download-details&file=74

**Comment by Portisch** [comment]
I will take a look if I find the source. I stopped using the receiver ~5 years ago.

**Comment by bladem2003** [comment]
Hello
I have the same problem after update to 1809.
it works like a charm for many years congratulation for your work.

**Comment by Mek7** [comment]
Hi,
I finished my software for Windows. It replaces Girder and the old DLL plugin. You can map remote control buttons to keyboard shortcuts.
The program is called MekIR and can be downloaded on my web page: http://mekweb.eu/?lang=en&q=download-details&file=74
Hope it helps.

**Comment by bladem2003** [comment]
The dll works with the same Windows version (1809) in a virtual machine.

**Comment by bladem2003** [comment]
@Portisch
Do you find the source?

**Comment by Portisch** [comment]
Yes, I also found a newer version 1.0.0.15. I have to cleanup and check the source before uploading.
The Source is for [RAD Studio](https://www.embarcadero.com/de/products/rad-studio) what I do not have anymore sou I can't compile a new version...

**Comment by bladem2003** [comment]
Thanks

**Comment by Portisch** [comment]
Can you try this DLL. I got the error before, but not after testing the new one.
Or it have a issue with permissions (INI file). Add maybe a "everyone" access to the Plugins folder.

**Comment by bladem2003** [comment]
Now I get the error "Device can not be opened". I have full rights to the folder. I have also tested it in a vmware with the 1809 and there is no error. strange

**Comment by Portisch** [comment]
It is not a bug from my side! :)
It is a bug from a component used in the source:
https://en.delphipraxis.net/topic/453-tjvhiddevicecontroller-throwing-errors-after-windows-10-1809-upgrade/

After compiling version 1.0.0.15 I used a newer version of the component and the message changed from null pointer error to cannot open.
I will try the hint described in the link above.

**Comment by Portisch** [comment]
Please try this version:
[USB_IR_Remote_Receiver_1.0.0.15.zip](https://github.com/Portisch/USB-IR-Remote-Receiver/files/2842417/USB_IR_Remote_Receiver_1.0.0.15.zip)

**Comment by bladem2003** [comment]
The last one works now with 1809.
Many thanks.
